packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -194,20 +206,27 @@ export class SOQLEditorInstance {
     }
   }
 
-  protected handleRunQuery(): Promise<void> {
+  protected async handleRunQuery(): Promise<void> {
     // Check to see if a default org is set.
     if (!workspaceContext.username) {
       // i18n
       const message = `No default org found. Set a default org to use SOQL Builder. Run "SFDX: Create a Default Scratch Org" or "SFDX: Authorize an Org" to set one.`;
       channelService.appendLine(message);
       vscode.window.showInformationMessage(message);
+      this.runQueryDone();
       return Promise.resolve();
     }
 
     const queryText = this.document.getText();
-    return withSFConnection(async conn => {
-      const queryData = await new QueryRunner(conn).runQuery(queryText);
-      this.openQueryDataView(queryData);
+    const conn = await workspaceContext.getConnection();
+    const queryData = await new QueryRunner(conn).runQuery(queryText);
+    this.openQueryDataView(queryData);
+    this.runQueryDone();
+  }
```

Comment:
@dehru It turns out that VSCode's `withProgress` can handle a single Promise chain, but nested callbacks mess it up. To fix the notification progress bar disappearing I had to unwind the call to `withSFConnection`.

Reply:
I think this one is ready for another QA attempt.